pkg/initializer/initializer.go

```diff
@@ -252,7 +258,15 @@ func (e *EtcdInitializer) restoreInMultiNode(ctx context.Context) error {
 		return fmt.Errorf("unable to remove the data-dir %v", err)
 	}
 
-	if err := retry.OnError(retry.DefaultBackoff, errors.AnyError, func() error {
+	if err := retry.OnError(retry.DefaultBackoff, errors.IsErrNotNil, func() error {
```

Comment:
This seems to be a code repetition. Can we extract this into a reusable function?